src/Yaapii.Atoms/IO/LoggingOutputStream.cs

```diff
@@ -0,0 +1,109 @@
+﻿// MIT License
+//
+// Copyright(c) 2020 ICARUS Consulting GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using Yaapii.Atoms.Scalar;
+
+namespace Yaapii.Atoms.IO
+{
+    /// <summary>
+    /// Logged output stream.
+    /// </summary>
+    public sealed class LoggingOutputStream : Stream
+    {
+        private readonly Stream origin;
+        private readonly string destination;
+        private readonly IScalar<IList<long>> bytes;
+        private readonly IScalar<IList<long>> time;
+
+        /// <summary>
+        /// Logged output stream.
+        /// </summary>
+        /// <param name="output">Destination of data</param>
+        /// <param name="destination">The name of source data</param>
+        public LoggingOutputStream(Stream output, string destination)
+        {
+            this.origin = output;
+            this.destination = destination;
+            this.bytes = 
+                new ScalarOf<IList<long>>(
```

Comment:
Why do you place scalars here? They are not necessary.

Reply:
I use a scalar because the values in the list are modified in the write methode. If I use a list instead of a scalar list I am not able to change the values in the list.

Reply:
Of course you are, you can change items in the list.